bluebutton-data-pipeline-fhir-load/src/main/java/gov/hhs/cms/bluebutton/datapipeline/fhir/transform/DataTransformer.java

```diff
@@ -724,16 +724,19 @@ private TransformedBundle transformCarrierClaim(RifRecordEvent<CarrierClaimGroup
 		 * updating them would require an extra roundtrip to the server (can't
 		 * think of an intelligent client-specified ID for them).
 		 */
-		Practitioner referrer = new Practitioner();
-		referrer.addIdentifier().setSystem(CODING_SYSTEM_NPI_US).setValue(claimGroup.referringPhysicianNpi);
-		Reference referrerReference = upsert(bundle, referrer,
-				referencePractitioner(claimGroup.referringPhysicianNpi).getReference());
-		ReferralRequest referral = new ReferralRequest();
-		referral.setStatus(ReferralStatus.COMPLETED);
-		referral.setPatient(referencePatient(claimGroup.beneficiaryId));
-		referral.addRecipient(referrerReference);
-		// Set the ReferralRequest as a contained resource in the EOB:
-		eob.setReferral(new Reference(referral));
+		if (claimGroup.referringPhysicianNpi.isPresent()) {
+			Practitioner referrer = new Practitioner();
+			referrer.addIdentifier().setSystem(CODING_SYSTEM_NPI_US)
+					.setValue(claimGroup.referringPhysicianNpi.toString());
```

Comment:
Isn't using `toString()` here (and, it looks like, on every other `Optional` a fairly major bug? I'm looking at the code for it, and it looks like it should be returning something like the following: `String.format("Optional[%s]", value)`.

If I'm right about that being a bug, I think this also illustrates a problem with your tests, that they didn't catch it. You should add at least a couple sanity checks in there that would verify this.

Even if I'm wrong and this just returns the bare value, it's more semantically correct to use `Optional.get()` on all of these.